community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/TraversalAStar.java

```diff
@@ -122,7 +122,8 @@ public WeightedPath findSinglePath( Node start, Node end )
                 new SelectorFactory( end, interest ) )
                 .evaluator( includeWhereEndNodeIs( end ) )
                 .traverse( start );
-        return () -> new WeightedPathIterator( lastTraverser.iterator(), costEvaluator, stopAfterLowestWeight );
+        return Iterators.asResourceIterable(
```

Comment:
Doesn't look like this transformation is needed since `WeightedPathIterator` is now a `ResourceIterator`